packages/framework-provider-aws/src/library/searcher-adapter.ts

```diff
@@ -32,62 +31,150 @@ export async function searchReadModel(
+        case 'gte':
+          return `#${propName} >= ${holder(index)}`
+        case 'lte':
+          return `#${propName} <= ${holder(index)}`
+        case 'in':
+          return `#${propName} IN (${value
+            .map((value: any, subIndex: number) => `${holder(index, subIndex)}`)
+            .join(',')})`
+        case 'contains':
+          return `contains(#${propName}, ${holder(index)})`
+        case 'beginsWith':
+          return `begins_with(#${propName}, ${holder(index)})`
+        case 'includes':
+          return `contains(#${propName}, ${holder(index)})`
+        default:
+          if (typeof value === 'object') {
+            return `#${propName}.${buildOperation(operation, value)}`
+          }
+          throw new InvalidParameterError(`Operator "${operation}" is not supported`)
+      }
+    })
+    .join(' AND ')
 }
 
-function placeholderBuilderFor(propName: string): (valueIndex: number) => string {
-  return (valueIndex: number) => `:${propName}_${valueIndex}`
+const attrCount: string[] = []
+function resetAttrCount(): void {
+  attrCount.length = 0
+}
+function placeholderBuilderFor(propName: string): (valueIndex: number, valueSubIndex?: number) => string {
+  return (valueIndex: number, valueSubIndex?: number) => {
+    const placeholder = `:${propName}_${valueIndex}` + (typeof valueSubIndex === 'number' ? `_${valueSubIndex}` : '')
+    if (attrCount.includes(placeholder)) return placeholderBuilderFor(propName)(valueIndex + 1)
+    attrCount.push(placeholder)
+    return placeholder
+  }
 }
 
-function buildExpressionAttributeNames(filters: Record<string, Filter<any>>): ExpressionAttributeNameMap {
+function buildExpressionAttributeNames(filters: FilterFor<any>): ExpressionAttributeNameMap {
   const attributeNames: ExpressionAttributeNameMap = {}
   for (const propName in filters) {
-    attributeNames[`#${propName}`] = propName
+    switch (propName) {
+      case 'not':
+        Object.assign(attributeNames, buildExpressionAttributeNames(filters[propName] as FilterFor<any>))
+        break
+      case 'and':
+      case 'or':
+        for (const filter of filters[propName] as Array<FilterFor<any>>) {
+          Object.assign(attributeNames, buildExpressionAttributeNames(filter as FilterFor<any>))
+        }
+        break
+      case 'includes':
+        break
+      default:
+        Object.entries(filters[propName] as FilterFor<any>).forEach(([prop, value]) => {
+          attributeNames[`#${propName}`] = propName
+          if (typeof value === 'object' && !Array.isArray(value)) {
+            Object.assign(attributeNames, buildExpressionAttributeNames({ [prop]: value }))
+          }
+        })
+        break
+    }
   }
   return attributeNames
 }
 
-function buildExpressionAttributeValues(filters: Record<string, Filter<any>>): ExpressionAttributeValueMap {
+function buildExpressionAttributeValues(filters: FilterFor<any>): ExpressionAttributeValueMap {
```

Comment:
Do the same changes that we did in "buildFilterExpression" regarding removing the attrCoung global array